src/components/radio/radio_button.ts

```diff
@@ -0,0 +1,298 @@
+    }
+  }
+
+  @Input()
+  get selected() {
+    return this._selected;
+  }
+
+  set selected(selected: MdRadioButton) {
+    this._selected = selected;
+    this.value = selected.value;
+
+    selected.checked = true;
+  }
+}
+
+
+@Component({
+  selector: 'md-radio-button',
+  templateUrl: './components/radio/radio_button.html',
+  styleUrls: ['./components/radio/radio-button.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdRadioButton {
+  @HostBinding('class.md-radio-focused')
+  private _isFocused: boolean;
+
+  /** Whether this radio is checked. */
+  private _checked: boolean = false;
+
+  /** The unique ID for the radio button. */
+  @HostBinding('id')
+  @Input()
+  id: string;
+
+  /** Analog to HTML 'name' attribute used to group radios for unique selection. */
+  @Input()
+  name: string;
+
+  /** Whether this radio is disabled. */
+  private _disabled: boolean;
+
+  /** Value assigned to this radio.*/
+  private _value: any = null;
+
+  /** The parent radio group. May or may not be present. */
+  radioGroup: MdRadioGroup;
+
+  /** Dispatcher for coordinating radio unique-selection by name. */
+  radioDispatcher: MdRadioDispatcher;
+
+  /** Event emitted when the group value changes. */
+  @Output()
+  change: EventEmitter<MdRadioChange> = new EventEmitter();
+
+  constructor(@Optional() radioGroup: MdRadioGroup, @Attribute('id') id: string,
+              radioDispatcher: MdRadioDispatcher) {
+    // Assertions. Ideally these should be stripped out by the compiler.
+    // TODO(jelbourn): Assert that there's no name binding AND a parent radio group.
+
+    this.radioGroup = radioGroup;
+    this.radioDispatcher = radioDispatcher;
+
+    this.id = id != null ? id : `md-radio-${_uniqueIdCounter++}`;
+
+    radioDispatcher.listen((name: string) => {
+      if (name == this.name) {
+        this.checked = false;
+      }
+    });
+  }
+
+  /*
+   * We use a hidden native input field to handle changes to focus state via keyboard navigation,
+   * with visual rendering done separately. The native element is kept in sync with the overall
+   * state of the component.
+   */
+  onInputFocus() {
+    this._isFocused = true;
+  }
+
+  onInputBlur() {
+    this._isFocused = false;
+  }
+
+  /** Input change handler, called only on keyboard selection. */
+  onInputChange() {
+    this.checked = true;
+  }
+
+  get inputId(): string {
+    return this.id + '-input';
+  }
+
+  @HostBinding('class.md-radio-checked')
+  @Input()
+  get checked(): boolean {
+    return this._checked;
+  }
+
+  set checked(value: boolean) {
+    if (value) {
+      // Notify all radio buttons with the same name to un-check.
+      this.radioDispatcher.notify(this.name);
+    }
+    this._checked = value;
+  }
+
+  /** MdRadioGroup reads this to assign its own value. */
+  @Input()
+  get value(): any {
+    return this._value;
+  }
+
+  set value(value: any) {
+    if (this._value != value) {
+      let event = new MdRadioChange();
```

Comment:
Sorry, don't know what I was thinking here, got this confused with checked.
I moved the code into checked, which matches the native input change event behavior (emit change only when selected).